truffle/contracts/Splitter.sol

```diff
@@ -0,0 +1,106 @@
+pragma solidity 0.5.10;
+
+import './Owned.sol';
+
+contract Splitter is Owned {
+    
+    struct Person {
+        address wallet;
+        bytes32 name;
+        uint    pendingWithdrawals;
+        uint    index;
+        bool    isParticipant;
+    }
+    
+    mapping(address => Person) participantStorage;
+    address[] participantList;
+    uint public balance;
+
+    event LogParticipantAdded(address participant, bytes32 name);
+    event LogPariticipantDeleted(address participant);
+    event LogEthSent(address sender, uint amount);
+    event LogWithdrawed(address receiver);
+    
+    modifier _onlyParticipant {
+        require(participantStorage[msg.sender].isParticipant);
+        _;
+    }
+    
+    constructor(bytes32 name) public payable {
+        balance = 0;
+        addParticipant(msg.sender, name);
+    }
+    
+    function addParticipant(address participant, bytes32 name) public _onlyOwner returns(bool) {
+        require(!participantStorage[participant].isParticipant);
+        
+        participantStorage[participant].wallet = participant;
+        participantStorage[participant].name = name;
+        participantStorage[participant].pendingWithdrawals = 0;
+        participantStorage[participant].index = participantList.length;
+        participantStorage[participant].isParticipant = true;
+        participantList.push(participant);
+
+        emit LogParticipantAdded(participant, name);
+        
+        return true;
+    }
+    
+    function deleteParticipant(address participant) public _onlyParticipant returns(bool) {
+        require(msg.sender == participant && msg.sender != owner);
+        
+        uint index = participantStorage[msg.sender].index;
+        
+        participantList[index] = participantList[participantList.length - 1];
+        participantStorage[participantList[participantList.length - 1]].index = index;
+        participantList.length--;
+
+        delete(participantStorage[msg.sender]);
+        
+        emit LogPariticipantDeleted(msg.sender);
+    }
+    
+    function getParticipants() public view returns(address[] memory, bytes32[] memory) {
+        address[] memory addrs = new address[](participantList.length);
+        bytes32[] memory names = new bytes32[](participantList.length);
+
+        for (uint i = 0; i < participantList.length; i++) {
```

Comment:
This unbounded `for` loop will run out of gas at `.length > something`. This is a problem because it means the contract could be subject to freezing up when it's successful. 

Some contracts do support a "batch" processing scheme where the iteration/batch size is tied to an input array. That gives users some latitude to figure out how big a batch can safely be. In this case, the function will just become unusable. 

An argument can be made for more reliance on event logs, i.e., the client should discover this another way, but data discoverability is a good thing. A way to make it scaleable is to present two functions. One to return a specific row (`i`), and another to return `.length`. The iteration concern is pushed out to clients who can loop away as much as they want.

Reply:
Ok, let me summarize this. You advise me to create two functions whose aims would be, respectively, to return the number of participants and to return a specific row. Consequently, the client must iterate on indexes and call the contract to retrieve a given participant. Would it not be too gas consuming? Imagine having 100+ participants in the Splitter (it should normally not happen but who knows ? ^^)